tests/src/Kernel/Form/AuthenticationFormTest.php

```diff
@@ -103,6 +97,65 @@ public function testGenerateNewAuthKey() {
     $this->assertSame(EdgeKeyTypeInterface::EDGE_AUTH_TYPE_BASIC, $decoded['auth_type']);
   }
 
+  /**
+   * Test private file path errors.
+   *
+   * @throws \Drupal\Core\Form\EnforcedResponseException
+   * @throws \Drupal\Core\Form\FormAjaxException
+   */
+  public function testPrivateFileSystemPathErrors() {
+    $form_state = new FormState();
+    $form = \Drupal::formBuilder()->buildForm(AuthenticationForm::class, $form_state);
+    $this->assertInstanceOf(AuthenticationForm::class, $form_state->getFormObject());
+
+    // Private file system is not configured.
+    $this->assertNotEmpty($form['connection_settings']['unconfigurable']['description']);
+    $this->assertTrue($form['actions']['#disabled']);
+    $this->assertEquals('The Drupal private file setting has not been configured.', $form['connection_settings']['unconfigurable']['label']['#value']);
+
+    // Add file_private_path setting, but not the private file dir.
+    $private_directory = DrupalKernel::findSitePath(Request::create('/')) . '/private';
+    $this->setSetting('file_private_path', $private_directory);
```

Comment:
Could not we use a path in the virtual fs for testing this instead of actually creating dirs and files on the hard drive?
See: `\Drupal\Tests\apigee_edge\Kernel\OauthTokenFileStorageTest` and `\Drupal\KernelTests\KernelTestBase::setUpFilesystem()` and `\Drupal\KernelTests\Core\File\FileTestBase::setUpFilesystem()`.

Reply:
Nice, thanks for info on virtual fs, updated.